internal/controller/apiextensions/composite/composition_pt.go

```diff
@@ -526,11 +526,11 @@ func (r *APIDryRunRenderer) Render(ctx context.Context, cp resource.Composite, c
 
 	for i := range t.Patches {
 		if err := Apply(t.Patches[i], cp, cd, patchTypesFromXR()...); err != nil {
-			return errors.Wrapf(err, errFmtPatch, i)
+			return errors.Wrapf(err, errFmtPatch, t.Patches[i].Type, i, t.Patches[i].GetFromFieldPath())
```

Comment:
Should we use `toFieldPath` instead of `fromFieldPath`, which feels more appropriate considering the error message 🤔

Reply:
And it should always be available, including the [`Combine*` type patches](https://docs.crossplane.io/v1.13/concepts/patch-and-transform/#types-of-patches).